Allow clicking on whitespace to right of categories and menu options (issue 729)

juzen2003 opened the pull request:
- Fixes #729 
- Were any Django, import pipeline, table_schema, or dictionary files modified? N
  - Database used: opus3_test_200107
  - All Django tests pass: Y
- Were any JavaScript or CSS files modified? Y
  - JSHINT run on all affected files: Y (menu.js)
  - Tested on Chrome, Firefox, Safari, and iOS: Y
    (Remember to test all browser sizes)
  - Tested for race conditions (with delayed API calls if applicable): NA

Description of changes:
- Make the whitespace to right of categories and menu options clickable in search sidebar menu and select metadata menu.

juzen2003 committed:
Make sure white space to the right of categories in select metadata menu
are clickable to expand/collapse each category.
- In opus.css (line 706-714), add the styling to make sure the a tags
  for categories take up full width in select metadata menu.

juzen2003 committed:
Make the whole row (whitespace) clickable for all categories and menu
options in search sidebar menu and select metadata menu.
- In menu.html (line 53,68,95,114), wrap the checkmark, i icon, and menu
  text into a span. It will be easier to set the background color for all
  them when they are grouped together.
- In opus.css (line 706-710), add the styling to make sure the a tags for
  categories take up full width in select metadata and search sidebar menu.
- In opus.css (line 1511), make sure each menu option takes up full width
  in select metadata and search sidebar menu.
- In opus.css (line 1562-1565), remove the padding to make sure the full
  whitespace is clickable for each menu option.
- In menu.js (line 95,102), make sure only the checkmark, i icon, and
  menu option text are highlighted/de-highlighted when a menu option is
  clicked.

juzen2003 committed:
Clean up & add comments in menu.js (line 92,97) & opus.css (line 1544,1549).

juzen2003 committed:
Vertically center categories in search sidebar menu and select metadata
menu. Also make padding clickable by adding padding to anchor tags inside
the li tag.
- In menu.html (line 4,7,9,33), remove padding styling from li and add
  padding styling to anchor tag. This will make padding area clickable
  in search sidebar menu and select metadata menu.

juzen2003 committed:
Make sure sub categories have 0 padding-top & padding-bottom.
- In menu.html (line 33), remove py-2 for sub categories in search sidebar
  menu and select metadata menu.

rfrenchseti commented on opus/application/static_media/css/opus.css:
```diff
@@ -703,6 +703,15 @@ button.op-mini-thumbnail-zoom {
 
+/* Make sure the a tags for categories take up full width in select metadata
+and search sidebar menu */
+a[href^="#selector-submenu-search"], a[href^="#selector-submenu-obs"],
```
Would it be better to add a class to these instead of using the ID?

juzen2003 replied:
Added a class "op-submenu-category" to these anchor tags.

juzen2003 committed:
Add op-submenu-category as the class name for the category anchor tags
in search sidebar and select metadata menu.
- In menu.html (line 7,9,33), add op-submenu-category as the class name
  for the category anchor tags.
- In opus.css (line 708), update the selector to .op-submenu-category
  for the styling that make the category a tags take up full width in
  search sidebar and select metadata menu.

juzen2003 committed:
Vertically center each search term in search sidebar and select metadata
menu by setting both margin-top and margin-bottom to the same value.
- In opus.css (line 1544-1547), set margin-top and margin-bottom to 1px
  for search term so that it's vertically centered.

juzen2003 committed:
Reduce the spacing between the category name and the first search term
under the category when it's expanded.
- In menu.html (line 16,17), add class op-submenu-item-list to the ul
  for search terms in search sidebar and select metadata menu.
- In menu.html (line 714-721), add styling to shift the whole search term
  list up to reduce the spacing between the category name and search term
  in search sidebar and select metadata menu.

juzen2003 committed:
Position the search sidebar spinner right next to "General Constraints".
- In menu.html (line 7-16), group the general constraints anchor tag with
  .op-menu-text spinner so that they stay in one line.
- In opus.css (line 706-723), add the styling to make sure .op-menu-text
  and "General Constraints" stay in one line, and when the spinner is
  spinning, it's located right next to "General Constraints".

rfrenchseti reviewed (changes requested):
There's too much whitespace above the first category in the Select Metadata dialog. Flip your tab from your new version of OPUS to the one on tools to see the difference. Also it seems that the fields are a little closer together vertically than they used to be.

juzen2003 committed:
Update the styling to reduce the white space between the first category
name and the border line above it in select metadata menu. Also match
the field height with the one in the current OPUS.
- In opus.css (line 783-787), make the first category name closer to the
  border line above it in select metadata menu.
- In opus.css (line 111-113), match the field height in search tab with
  the one in the current OPUS.

rfrenchseti reviewed (changes requested):
I have no idea why, but now in the Select Metadata dialog, when you expand a category on the left, the scrollbar height doesn't update. When you nudge the scrollbar just a tiny bit, then it updates.

juzen2003 committed:
Update the regex to capture the DOM changes for selectMetadataMenuObserver.
This will make sure PS in "Available Metadata Fields" is properly updated
when category is expanded.
- In mutationObserver.js (line 163), update the regex pattern to detect
  the last DOM change happened in select metadata menu.
